awk/komand_awk/util/utils.py

```diff
@@ -1,21 +1,22 @@
 import komand
 import base64
 
-
 # Specify path for unix utils grep and wc:
 awk = "gawk"
 
+
 # Preprocess expression
-def preprocess_expression(log, expression):
-    if expression[0].isalnum()\
-       or expression.startswith('=')\
-       or expression.endswith('='):
-        expression = base64.b64decode(expression)
+def preprocess_expression(expression):
+    if expression[0].isalnum() \
+            or expression.startswith('=') \
+            or expression.endswith('='):
+        expression = base64.b64decode(expression).decode('utf-8')
     return expression
 
+
 # Run awk on text for pattern, return stdout
 def process_lines(log, text, expression):
-    cmd = 'echo "%s" | %s %s' % (text, awk, expression)
+    cmd = 'echo "{}" | {} {}'.format(text, awk, expression)
```

Comment:
.format is fine and much better then %, but checkout. f strings
https://realpython.com/python-f-strings/

You can keep it as format just want you to know about them Luke turned me onto them and I use them all the time now.

Reply:
Ok. I will use it from next amendments.